internal/uvm/vnuma.go

```diff
@@ -0,0 +1,197 @@
+//go:build windows
+
+package uvm
+
+import (
+	"fmt"
+
+	hcsschema "github.com/Microsoft/hcsshim/internal/hcs/schema2"
+)
+
+// prepareVNumaTopology creates vNUMA settings for implicit (platform) or explicit (user-defined) topology.
+//
+// For implicit topology we look for `MaxProcessorsPerNumaNode` and `MaxSizePerNode` create options. Setting them
+// in HCS doc, will trigger platform to create vNUMA topology based on the given values. Based on experiments, the
+// platform will create an evenly distributed topology based on requested memory and processor count for the HCS VM.
+//
+// For explicit topology we look for `NumaMappedPhysicalNodes`, `NumaProcessorCounts` and `NumaMemoryBlocksCounts` create
+// options. The above options are number slices, where a value at index `i` in each slice represents the corresponding
+// value for the `i`th vNUMA node.
+// Limitations:
+// - only hcsschema.MemoryBackingType_PHYSICAL is supported
```

Comment:
aren't we also setting virtual memory backing type above? Should we just remove that?

Reply:
Yes, done.